File: test/Data.Analyzers.Vsix.Test/CSharp/MissingDbMockFactoryMethod.cs
using DevZest.Data.Primitives;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    public class MyMock : DbMock<DbSession>
    {
        protected override void Initialize()
        {
        }
    }

    public class MyMockWithoutWarning : DbMock<DbSession>
    {
        public static Task<DbSession> CreateAsync(DbSession db, IProgress<DbGenerationProgress> progress = null, CancellationToken ct = default(CancellationToken))
        {
            return new MyMockWithoutWarning().MockAsync(db, progress, ct);
        }

        protected override void Initialize()
        {
        }
    }
}

File: test/Data.Analyzers.Vsix.Test/VisualBasic/MissingDbMockFactoryMethod.vb
Imports System.Threading
Imports DevZest.Data.Primitives

Public Class MyMock
    Inherits DbMock(Of DbSession)

    Protected Overrides Sub Initialize()
    End Sub
End Class

Public Class MyMockWithoutWarning
    Inherits DbMock(Of DbSession)

    Public Shared Function CreateAsync(db As DbSession, Optional progress As IProgress(Of DbInitProgress) = Nothing, Optional ct As CancellationToken = Nothing) As Task(Of DbSession)
        Return New MyMockWithoutWarning().MockAsync(db, progress, ct)
    End Function

    Protected Overrides Sub Initialize()
    End Sub
End Class
